### PR TITLE
Migrate keda k3d jobs

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- do not create VM for k3d test scenarios (use docker-in-docker logic)

**Related issue(s)**

https://github.com/kyma-project/keda-manager/issues/210

### CODE REVIEW

File: prow/jobs/modules/external/keda-manager.yaml [hunk 1]
```diff
@@ -161,9 +161,6 @@ presubmits: # runs on PRs
           - image: "eu.gcr.io/sap-kyma-neighbors-dev/image-builder:v20230313-8dfce5f0b"
             securityContext:
               privileged: false
-              seccompProfile:
```
[the remark this answers is not in the document]
Comment (pPrecel) [reply prow/jobs/modules/external/keda-manager.yaml]: @halamix2 thanks! done

**Review by halamix2 — APPROVED** [review]

**Pull request merged**